scripts/layerops.py

```diff
@@ -0,0 +1,721 @@
+            for entry in data:
+                temp.append(entry._get('techniques'))
+            self.mode = 'list'
+            self._merge_to_template()
+        else:
+            raise InvalidFormat
+        self.da = temp
+        self.corpus = self._build_template(temp)
+
+        # Default values for fields if they aren't supplied
+        self.default_values = {
+            "comment": "",
+            "enabled": True,
+            "color": "#ffffff",
+            "score": 1,
+            "metadata": []
+        }
+        self.processed = None
+
+    """
+        Processes the ingested data using User provided Lambdas
+        :param score: lambda to calculate score
+        :param comment: lambda to generate comments
+        :param enabled: lambda to determine enabled status
+        :param metadata: lambda to generate metadata
+        :param defaults: dictionary containing desired default values for missing data element values
+        :param out_Name: new name to apply to the resulting layer
+        :param out_Desc: new description to apply to the resulting layer
+    """
+    def process(self, score=None, comment=None, enabled=None, colors=None, metadata=None, defaults=None,
+                out_Name=None, out_Desc=None):
+        composite = copy.deepcopy(self.corpus)
+        if score != None:
+            for entry in composite:
+                entry['score'] = self._applyOperation(self.da, entry, 'score', score, defaults)
+
+        if comment != None:
+            for entry in composite:
+                entry['comment'] = self._applyOperation(self.da, entry, 'comment', comment, defaults)
+
+        if enabled != None:
+            for entry in composite:
+                entry['enabled'] = self._applyOperation(self.da, entry, 'enabled', enabled, defaults)
+
+        if colors != None:
+            for entry in composite:
+                entry['color'] = self._applyOperation(self.da, entry, 'color', colors, defaults)
+
+        if metadata != None:
+            for entry in composite:
+                entry['metadata'] = self._applyOperation(self.da, entry, 'metadata', metadata, defaults)
+
+        processed = copy.deepcopy(self.out)
+        processed['techniques'] = composite
+        if out_Name != None:
+            processed['name'] = out_Name
+        if out_Desc != None:
+            processed['description'] = out_Desc
+        self.processed = Layer(processed)
+
+    """
+        Retrieve processed results
+        :returns: A Layer object representing the processing changes
+    """
+    def get_processed(self):
+        if self.processed is not None:
+            return self.processed
+        else:
+            print('Please call the Process method to combine loaded layers.')
+
+    """
+        Edits name and description metadata on a processed layer
+        :param name: the new name to use
+        :param desc: the new description to use
+    """
+    def edit_meta(self, name=None, desc=None):
```

Comment:
The usage of `meta` is misleading here because metadata is a concept already in layer files. Might want to change to "information" or something to that effect, since that's what the controls group is called on the Navigator already.

Since out_name and out_desc is already a param for the process process, I'm not sure this is super useful to have as an independant function.

Reply:
This is gone in the next build.